src/UiControl.cc

```diff
@@ -1,14 +1,39 @@
-#include "../ui.h"
-#include "ui-node.h"
+#include <map>
 #include "nbind/api.h"
-#include "nbind/nbind.h"
+#include "control.h"
+#include "ui.h"
+
+std::map<uiControl *, UiControl *> controlsMap;
 
 uiControl *UiControl::getHandle() {
 	return handle;
 }
 
+static void _uicontrol_onDestroy(uiControl *control) {
+	UiControl *wrapper = controlsMap[control];
+	wrapper->onDestroy(control);
+	wrapper->originalDestroy(control);
+	wrapper->destroyed = true;
```

Comment:
This instance variable will be used to make every method noop when the libui control is  destroyed